qcengine/programs/nwchem/harvester.py

```diff
@@ -660,12 +662,28 @@ def harvest_outfile_pass(outtext):
     return psivar, psivar_coord, psivar_grad, version, error
 
 
-def harvest_hessian(hess):
-    """Parses the contents *hessian* of the NWChem hess file into a hessian array.
-    Hess file name has to be "nwchem.hess". (default)
+def harvest_hessian(hess: str) -> np.ndarray:
+    """Parses the contents of the NWChem hess file into a hessian array.
+
+    Args:
+        hess (str): Contents of the hess file
 
     """
-    raise NotImplementedError()
+
+    # Change the "D[+-]" notation of Fortran output to "E[+-]" used by Python
+    hess_conv = hess.replace("D", "E")
+
+    # Parse all of the float values
+    hess_triu = [float(x) for x in hess_conv.strip().splitlines()]
+
+    # The value in the Hessian matrix is the upper triangle
+    #  We first convert it to a 2D array
+    n = int(np.sqrt(8 * len(hess_triu) + 1) - 1) // 2  # Size of the 2D matrix
+    hess_arr = np.zeros((n, n))
+    hess_arr[np.triu_indices(n)] = hess_triu
```

Comment:
You can do `hess_arr[np.tril_indices(n, -1)] = hess_triu`

Reply:
Ah whoops this is actually

```python
idx = np.tril_indices(n, -1)
hess_arr[idx] = hess_arr.T[idx]
```

Reply:
Oh, that is much smarter! :D

Reply:
Okay, I had this wrong originally. TL;DR: Going to fix it. Long version...

After some further digging through NWChem source code, turns out the ".hess" file is the [lower matrix](https://github.com/nwchemgit/nwchem/blob/9d605c61bc3d4f18856c352f59ba0a5e75b0971b/src/stepper/stpr_gen_aux.F#L189).

`tril_indices` gives the correct behavior of having the "column" on the inner loop (0,0 -> 1,0 -> 1,1), whereas the `triu_indices` is equivalent to stepping over rows on the inner loop (0,0 -> 1,0 -> 2,0). 

Accordingly, the correct solution is similar to what I originally had but replacing `triu_indices` with `tril_indices`.